src/js/data/cosmos/index.ts

```diff
@@ -0,0 +1,79 @@
+import { of, throwError } from "rxjs";
+import { map, retry, switchMap } from "rxjs/operators";
+import { makeExecutableSchema } from "graphql-tools";
+import { RequestResponse } from "@dcos/http-service";
+import { CosmosClient, PackageVersionsResponse } from "cosmos-client";
+
+import Config from "#SRC/js/config/Config";
+import { Package, PackageSchema } from "#SRC/js/data/cosmos/Package";
+import { PackageVersionSchema } from "#SRC/js/data/cosmos/PackageVersion";
+
+export interface ResolverArgs {
+  cosmosClient: CosmosClient;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface PackageQueryArgs {
+  name: string;
+}
+
+function isPackageQueryArgs(args: GeneralArgs): args is PackageQueryArgs {
+  return (args as PackageQueryArgs).name !== undefined;
+}
+
+export const resolvers = ({ cosmosClient }: ResolverArgs) => ({
+  Package: {
+    versions(parent: { name: string }) {
+      if (!parent.name) {
+        return throwError("Package name must be available to resolve versions");
+      }
+
+      return of({}).pipe(
+        switchMap(() => cosmosClient.listPackageVersions(parent.name)),
```

Comment:
Seems indicative of something wrong that this can't start with `cosmosClient.listPackageVersions(parent.name).pipe(....`, still stumped by this

Reply:
Me too, I moved it around but this was they way the retry would function as desired.